src/stable_marriage.rs

```diff
@@ -1,96 +1,234 @@
-struct Individual {
-    cnt_match: Option<usize>,
+use itertools::Itertools;
+use std;
+use std::cmp::Ordering;
 
-    // For efficiency: most preferred last for men, most preferred first for women
-    is_man: bool,
-    prefs: Vec<usize>,
-}
+pub trait Matcher {
+    type Item;
+    type Target;
```

Comment:
In this file there is currently a mix between `Item` / `Target` and `Man` / `Woman`. If I understand correctly (and the need for me to say that proves the issue 😛), `Item` is basically `Man` (minus some computed metadata) and `Target` is `Woman` (minus some computed metadata).

I think it'd be better to rename these elements based on this fact, if I'm not mistaken?

Reply:
No, both Item and Target can be Man or Woman. The algorithm uses two matchers, one where Item=Man and Target=Woman, and another that has those two swapped. The naming is on purpose different

Reply:
Hmm OK. I guess I'll have to re-read the code to understand how it works then, thanks :)